### PR TITLE
Build tool content

### PR DESCRIPTION
**Pull request by stubailo**
Ready for review
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/meteor/guide/pull/111%23issuecomment-162645836%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23issuecomment-162701848%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23issuecomment-162789869%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23issuecomment-162801783%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46868728%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46868929%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46869231%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46869422%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46870053%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46870270%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46919152%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46919510%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46922567%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46922595%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46922606%22%2C%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46922661%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23issuecomment-162645836%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20ready%20for%20review%2C%20going%20to%20have%20some%20conversations%20with%20Ben%20and%20Uri%20about%20some%20of%20the%20XXXs%22%2C%20%22created_at%22%3A%20%222015-12-07T20%3A10%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22OK%20Uri%20will%20contribute%20some%20of%20the%20other%20parts%2C%20I%27d%20say%20we%20should%20review%20without%20them%20and%20get%20them%20in%20later.%22%2C%20%22created_at%22%3A%20%222015-12-07T23%3A15%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20looks%20great%2C%20I%20don%27t%20have%20much%20to%20say%20about%20it%22%2C%20%22created_at%22%3A%20%222015-12-08T06%3A39%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-12-08T07%3A38%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20e04552e8ae02d4004646abb59f3259fadc408801%20content/build-tool.md%2044%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46868929%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Mention%20reproducible%20builds%20and%20%60.meteor/versions%60%22%2C%20%22created_at%22%3A%20%222015-12-07T19%3A59%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-07T20%3A08%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20content/build-tool.md%3AL1-229%22%7D%2C%20%22Pull%20b7ad63df30535b4915be0624935f1c1cf4e89c88%20content/build-tool.md%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46919152%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20about%20%60meteor%20--production%60%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T06%3A31%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22That%27s%20mentioned%20in%20the%20minification%20section.%20I%20don%27t%20think%20it%20really%20helps%20you%20debug%20any%20production%20errors%20except%20those%20related%20to%20minification%2C%20so%20IDK%20if%20it%20needs%20a%20big%20mention%20here%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T07%3A37%3A56Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22Gonna%20merge%20anyway%2C%20LMK%20if%20that%27s%20not%20OK%22%2C%20%22created_at%22%3A%20%222015-12-08T07%3A38%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-12-08T07%3A38%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20content/build-tool.md%3AL1-233%22%7D%2C%20%22Pull%20e04552e8ae02d4004646abb59f3259fadc408801%20content/build-tool.md%2071%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46869231%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22One%20difference%20between...%20takes%20two%20arguments%22%2C%20%22created_at%22%3A%20%222015-12-07T20%3A01%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-07T20%3A03%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20content/build-tool.md%3AL1-229%22%7D%2C%20%22Pull%20e04552e8ae02d4004646abb59f3259fadc408801%20content/build-tool.md%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46868728%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22What%20is%20it%3F%3F%20It%27s%20the%20thing%20that%20runs%20when%20you%20type%20%60meteor%60%22%2C%20%22created_at%22%3A%20%222015-12-07T19%3A58%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-07T20%3A10%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20content/build-tool.md%3AL1-229%22%7D%2C%20%22Pull%20b7ad63df30535b4915be0624935f1c1cf4e89c88%20content/build-tool.md%20224%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/meteor/guide/pull/111%23discussion_r46919510%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20it%27s%20Ok%20to%20link%20to%20it%3F%22%2C%20%22created_at%22%3A%20%222015-12-08T06%3A39%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/132554%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tmeasday%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-08T07%3A37%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/448783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/stubailo%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20content/build-tool.md%3AL1-233%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull e04552e8ae02d4004646abb59f3259fadc408801 content/build-tool.md 7'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#discussion_r46868728'>File: content/build-tool.md:L1-229</a></b>
- <a href='https://github.com/stubailo'></a> What is it?? It's the thing that runs when you type `meteor`
- [x] <a href='#crh-comment-Pull e04552e8ae02d4004646abb59f3259fadc408801 content/build-tool.md 44'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#discussion_r46868929'>File: content/build-tool.md:L1-229</a></b>
- <a href='https://github.com/stubailo'></a> Mention reproducible builds and `.meteor/versions`
- [x] <a href='#crh-comment-Pull e04552e8ae02d4004646abb59f3259fadc408801 content/build-tool.md 71'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#discussion_r46869231'>File: content/build-tool.md:L1-229</a></b>
- <a href='https://github.com/stubailo'></a> One difference between... takes two arguments
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#issuecomment-162645836'>General Comment</a></b>
- <a href='https://github.com/stubailo'></a> Should be ready for review, going to have some conversations with Ben and Uri about some of the XXXs
- <a href='https://github.com/stubailo'></a> OK Uri will contribute some of the other parts, I'd say we should review without them and get them in later.
- <a href='https://github.com/tmeasday'></a> This looks great, I don't have much to say about it
- [x] <a href='#crh-comment-Pull b7ad63df30535b4915be0624935f1c1cf4e89c88 content/build-tool.md 25'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#discussion_r46919152'>File: content/build-tool.md:L1-233</a></b>
- <a href='https://github.com/tmeasday'></a> What about `meteor --production`?
- <a href='https://github.com/stubailo'></a> That's mentioned in the minification section. I don't think it really helps you debug any production errors except those related to minification, so IDK if it needs a big mention here?
- <a href='https://github.com/stubailo'></a> Gonna merge anyway, LMK if that's not OK
- [x] <a href='#crh-comment-Pull b7ad63df30535b4915be0624935f1c1cf4e89c88 content/build-tool.md 224'></a> &nbsp;<b><a href='https://github.com/meteor/guide/pull/111#discussion_r46919510'>File: content/build-tool.md:L1-233</a></b>
- <a href='https://github.com/tmeasday'></a> I think it's Ok to link to it?

<a href='https://www.codereviewhub.com/meteor/guide/pull/111?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/meteor/guide/pull/111?approve=1'></a>&nbsp;<a href='https://github.com/meteor/guide/pull/111'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by stubailo** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/meteor/guide/pull/111'>Refresh</a>.<a href='#crh-update-none'></a>

File: content/build-tool.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+title: Build system
+---
+
+## What does it do?
+
+The Meteor build tool is what compiles, runs, deploys, and publishes all of your Meteor apps and packages. It's Meteor's built-in solution to the problems also solved by tools like Grunt, Gulp, Webpack, Browserify, Nodemon, and many others, and uses many popular Node.js tools like Babel and UglifyJS internally to enable a seamless experience.
```
Comment (stubailo) [inline content/build-tool.md]: What is it?? It's the thing that runs when you type `meteor`

File: content/build-tool.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+[Atmosphere](https://atmospherejs.com/) is a repository and discovery website for Meteor-specific packages. Packages are published on Atmosphere when they need to take advantage of features specific to Meteor, like the cross-platform build system, isomorphic client/server code, or data system.
+
+#### Adding packages from Atmosphere
+
+You have two options for adding packages from Atmosphere to your app:
+
+1. Use the command line: `meteor add kadira:flow-router`
+2. Edit the file in your app under `.meteor/packages`, and add the package name anywhere in the file
+
+These options will add the newest version of the desired package that is compatible with the other packages in your app. If you want to specify a particular version, you can specify it by adding a suffix to the package name, like so: `meteor add kadira:flow-router@2.10.0`.
```
Comment (stubailo) [inline content/build-tool.md]: Mention reproducible builds and `.meteor/versions`

File: content/build-tool.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+#### Overriding packages from Atmosphere with a local version
+
+A Meteor app can load packages in one of two ways:
+
+1. Downloading a pre-built package from Atmosphere. The package is cached in `~/.meteor/packages` on Mac/Linux or `%LOCALAPPDATA%/.meteor/packages`, and only loaded into your app as it is built.
+2. Loading the package's source code into a `packages/` directory inside your app. This lets you modify the source code of the package for your particular needs.
+
+If you need to patch a package to do something that the published version doesn't do, then (2) is the option for you. You can even do this to load patched versions of Meteor core packages - just copy the code of the package from [Meteor's GitHub repository](https://github.com/meteor/meteor/tree/devel/packages), and edit away.
+
+One difference between pre-published packages is that the published packages have any binary dependencies pre-built. This should only affect a small subset of packages. If you clone the source code into your app, you need to make sure you have any compilers required by that package pre-installed.
```
Comment (stubailo) [inline content/build-tool.md]: One difference between... takes two arguments

Comment (stubailo) [inline content/build-tool.md]: :+1:

File: content/build-tool.md (hunk 2 above)
Comment (stubailo) [inline content/build-tool.md]: :+1:

File: content/build-tool.md (hunk 1 above)
Comment (stubailo) [inline content/build-tool.md]: :+1:

**Comment by stubailo** [comment]
Should be ready for review, going to have some conversations with Ben and Uri about some of the XXXs

**Comment by stubailo** [comment]
OK Uri will contribute some of the other parts, I'd say we should review without them and get them in later.

File: content/build-tool.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+### Combines and minifies code
+
+Another important feature of the Meteor build tool is that it automatically concatenates and minifies all of your files in production mode. This is enabled by the `standard-minifiers` pacakge, which is in all Meteor apps by default. If you need different minification behavior, you can replace this package. Below, we'll talk about how to switch out your minifier to add PostCSS to your build process.
+
+### Development vs. production
+
+Running an app in development is all about fast iteration time. All kinds of different parts of your app are handled differently and instrumented to enable better reloads and debugging. In production, the app is reduced to just the necessary code, and functions like a regular Node.js app. Therefore, you shouldn't run your app in production by running the `meteor` command. Instead, run `meteor build` and then deploy the resulting app bundle. Read more in the [production deployment article](XXX).
```
Comment (tmeasday) [inline content/build-tool.md]: What about `meteor --production`?

File: content/build-tool.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+## Using community packages
+
+Building an application completely from scratch is a tall order. This is one of the main reasons you might consider using Meteor in the first place - you can focus on writing the code that is specific to your app, instead of reinventing wheels like user login and data synchronization. To streamline your workflow even further, it makes sense to use community packages from Atmosphere and NPM. Many of these packages are recommended in the guide, and you can find more in the online directories.
```
Comment (tmeasday) [inline content/build-tool.md]: Links please ;)

File: content/build-tool.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+Currently, Meteor doesn't have a separate build step for post-processing CSS, so the only way to integrate it is to build it into the minifier. Thankfully, there is a community package that has integrated PostCSS with plugin support into a replacement for Meteor's standard minification package.
+
+### juliancwirko:postcss
+
+Use the package [juliancwirko:postcss](https://atmospherejs.com/juliancwirko/postcss) to your app to enable PostCSS for your Meteor app. It's not completely trivial to set it up, and we hope to make support for PostCSS a more core part of Meteor in the future. Read the documentation for the package to get the steps to add it to your app; we won't reproduce them here since they might change in future versions.
+
+// XXX do we want to do the above? Or should we reproduce the directions?
```
Comment (tmeasday) [inline content/build-tool.md]: I think it's Ok to link to it?

**Comment by tmeasday** [comment]
This looks great, I don't have much to say about it

File: content/build-tool.md (hunk 6 above)
Comment (stubailo) [inline content/build-tool.md]: :+1:

File: content/build-tool.md (hunk 4 above)
Comment (stubailo) [inline content/build-tool.md]: That's mentioned in the minification section. I don't think it really helps you debug any production errors except those related to minification, so IDK if it needs a big mention here?

Comment (stubailo) [inline content/build-tool.md]: Gonna merge anyway, LMK if that's not OK

Comment (stubailo) [inline content/build-tool.md]: :+1:<a href='#crh-update-complete'></a>

**Comment by stubailo** [comment]
:+1:<a href='#crh-update-none'></a>

**Pull request merged**